### PR TITLE
SDCORE-865: Timestamp updte for each NF profile

### PR DESCRIPTION
**Pull request by vthiruveedula**

### CODE REVIEW

File: producer/nf_discovery.go [hunk 1]
```diff
@@ -95,6 +96,17 @@ func NFDiscoveryProcedure(queryParameters url.Values) (response *models.SearchRe
 		logger.DiscoveryLog.Warnln("NF Profile Raw decode error: ", nfProfilesStruct)
 	}
 
+	// sort nfprofiles based on timestamp
+	sort.Slice(nfProfilesStruct, func(i, j int) bool {
+		var updatedTimeVal time.Time
+		if nfProfilesStruct[j].CustomInfo == nil {
+			return false
+		}
+		updatedTimeVal = nfProfilesStruct[j].CustomInfo["updatedAt"].(time.Time)
```
Comment (badhrinathpa) [inline producer/nf_discovery.go:105]: where do we fill this custom info field ? should we update it to use createdAt field ?

File: go.mod [hunk 2]
```diff
@@ -2,15 +2,16 @@ module github.com/omec-project/nrf
 
 go 1.14
 
+replace github.com/omec-project/nrf => ./
```
Comment (badhrinathpa) [inline go.mod:5]: what is this needed for ? dbadapter ?

File: dbadapter/dbadapter.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+func ConnectToDBClient(setdbName string, url string, enableStream bool, nfProfileExpiryEnable bool) DBInterface {
+	dbc := &MongoDBClient{}
+	MongoDBLibrary.SetMongoDB(setdbName, url)
+	if MongoDBLibrary.Client != nil {
+		dbc.Client = MongoDBLibrary.Client
+		dbc.dbName = setdbName
+	}
+	DBClient = dbc
+
+	if enableStream {
```
Comment (badhrinathpa) [inline dbadapter/dbadapter.go:59]: lets add a task to change configuration dynamically. if we disable this flag, we would need to close the stream like NfProfStream.Close()

File: go.mod (hunk 2 above)
Comment (vthiruveedula) [reply go.mod:5]: yes, its required for dbadaptor

File: dbadapter/dbadapter.go (hunk 3 above)
Comment (vthiruveedula) [reply dbadapter/dbadapter.go:59]: I will open a Jira ticket to enable or disable this flag dynamically with configmap

**Review by badhrinathpa — APPROVED** [review]
+1

**Pull request merged**